src/js/osweb/items/inline_javascript.js

```diff
@@ -0,0 +1,92 @@
+import Item from './item.js'
+import JavaScriptWorkspace from '../classes/javascript_workspace.js'
+
+/**
+ * Class representing an inline item.
+ * @extends Item
+ */
+export default class InlineJavaScript extends Item {
+  /**
+     * Create an inline item which executes inline python code.
+     * @param {Object} experiment - The experiment item to which the item belongs.
+     * @param {String} name - The unique name of the item.
+     * @param {String} script - The script containing the properties of the item.
+     */
+  constructor (experiment, name, script) {
+    // Inherited create.
+    super(experiment, name, script)
+    // Define and set the public properties.
+    this.description = 'Executes JavaScript code (ECMA 5.1)'
+    this.workspace = new JavaScriptWorkspace(this.experiment)
+    // Process the script
+    console.log(this.vars)
+    this.from_string(script)
+  }
+
+  /** Reset all item variables to their default value. */
+  reset () {
+    this.vars._prepare = ''
+    this.vars._run = ''
+  }
+
+  /**
+     * Parse a definition string and retrieve all properties of the item.
+     * @param {String} script - The script containing the properties of the item.
+     */
+  from_string (script) {

```

Comment:
@shyras I'm not so happy with the implementation of `from_script()`, because it treats the `_run` and `_prepare` variables as though they are special cases, which they arent: every item should be able to parse multiline variables of any name.

Reply:
Ok. That is something which we should look into. I thought that it were special cases. So far as I can see looking at the code, Daniel made some changes to the routine. Maybe he knows a quick fix, otherwise we should place it on the to-do list.

Reply:
But this is a whole new class, right? The one that Sebastiaan just added. The `from_script()` function is not here, or do you mean the `from_string()` function?

Reply:
Sorry, I mean the `from_string()` function indeed. I copied that from `inline_script.js`. It's not important for now, but the multiline-variable parsing should ideally be carried into the general functionality for parsing variables, which I assume happens in `item.js`?